dsa-23au/java-dsa/pswish-natmcl/pswish-app/src/main/java/com/pswishcorp/app/Car.java

```diff
@@ -4,16 +4,37 @@ public class Car {
     String cartype;
     String driver;
     int hwyMPG;
+    String carMake;
+    String carModel;
+    String fuelType;
+    String availabilityStatus;
+    double averageCostPerGallonPrem;
 
-    public Car(String driver, String cartype, int hwyMPG) {
+    public Car(String driver, String cartype, int hwyMPG, double averageCostPerGallonPrem) {
         this.cartype = cartype;
         this.hwyMPG = hwyMPG;
-        float averageCostPerGallonPrem;
+        this.averageCostPerGallonPrem = averageCostPerGallonPrem;
     }
 
     public static void main(String[] args) {
-        Car car1 = new Car("Paul","Q5", 27);
-        Car car2 = new Car("Nate", "Genesis", 28);
+        Car car1 = new Car("Paul","Suv", 27, .16);
+        Car car2 = new Car("Nate", "Sedan", 28,  .17);
         }
+
+    public Object getDriver() throws Exception {
+        throw new Exception("Not yet implemented, v0.4");
```

Comment:
I am not super familiar with the 'throw' function. what are its uses/advantages?